src/components/Registration/Manual/SearchAttendee.vue

```diff
@@ -1,148 +1,269 @@
 <script setup>
-import { ref, watch } from 'vue'
-import { XCircleIcon, MagnifyingGlassIcon, PrinterIcon } from '@heroicons/vue/20/solid'
+import { ref, watch, onMounted } from 'vue'
+import { XCircleIcon, PrinterIcon } from '@heroicons/vue/20/solid'
+import { FunnelIcon } from '@heroicons/vue/24/outline'
 
-const query = ref('')
-const typing = ref(false)
+// INITIALISE TEMPLATE REFS
 const searchBar = ref(null)
-
-const erase = () => {
-  query.value = ''
+const refs = ref([])
+onMounted(() => {
+  refs.value = Array(showInfo.length)
+    .fill(null)
+    .map(() => ref(null))
+})
+function setRef(index) {
+  return (el) => {
+    refs.value[index] = el
+  }
 }
 
+const menuOpen = ref(false)
+
+const toggleMenu = () => (menuOpen.value = !menuOpen.value)
+
+const query = ref('')
+const typing = ref(false)
+
 watch(query, () => {
   if (query.value !== '') typing.value = true
   else typing.value = false
 })
 
-const checkIn = (item) => {
-  item.checkedIn.value = true
-}
-
 const items = [
   {
     id: 1,
     name: 'Wei Tat Chung',
-    position: 'Chairperson',
-    role: 'Organiser',
     email: 'wtc@email.com',
-    organisation: 'SUSS AI-IG',
-    checkedIn: ref(false)
+    checkedIn: ref(false),
+    info: {
+      role: 'Chairperson',
+      memberType: 'Organiser',
+      organisation: 'SUSS AI-IG'
+    }
   },
   {
     id: 2,
-    name: 'Shaun Laclemence',
-    position: 'Chairperson',
-    role: 'Organiser',
+    name: 'Don Chia',
     email: 'wtc@email.com',
-    organisation: 'SUSS AI-IG',
-    checkedIn: ref(false)
+    checkedIn: ref(false),
+    info: {
+      role: 'Chairperson',
+      memberType: 'Organiser',
+      organisation: 'SUSS AI-IG'
+    }
   },
   {
     id: 3,
-    name: 'Don Chia',
-    position: 'Chairperson',
-    role: 'Organiser',
+    name: 'Shaun Ming Laclemence',
     email: 'wtc@email.com',
-    organisation: 'SUSS AI-IG',
-    checkedIn: ref(false)
+    checkedIn: ref(false),
+    info: {
+      role: 'Chairperson',
+      memberType: 'Organiser',
+      organisation: 'SUSS AI-IG'
+    }
   },
   {
-    id: 3,
-    name: 'Don Chia',
-    position: 'Chairperson',
-    role: 'Organiser',
+    id: 4,
+    name: 'Very very very very very very very very very very long name',
     email: 'wtc@email.com',
-    organisation: 'SUSS AI-IG',
-    checkedIn: ref(false)
+    checkedIn: ref(false),
+    info: {
+      role: 'Chairperson',
+      memberType: 'Organiser',
+      organisation: 'SUSS AI-IG'
+    }
   },
   {
-    id: 3,
-    name: 'Don Chia',
-    position: 'Chairperson',
-    role: 'Organiser',
+    id: 5,
+    name: 'Wei Tat Chung',
     email: 'wtc@email.com',
-    organisation: 'SUSS AI-IG',
-    checkedIn: ref(false)
+    checkedIn: ref(false),
+    info: {
+      role: 'Chairperson',
+      memberType: 'Organiser',
+      organisation: 'SUSS AI-IG'
+    }
   }
 ]
+
+const showInfo = [
```

Comment:
is this filterTypes? naming convention for data should be explaining its data; actions then actions